src/ui/public/doc_table/components/table_header.html

```diff
@@ -1,7 +1,16 @@
 <tr>
   <th width="1%"></th>
   <th ng-if="indexPattern.timeFieldName" data-test-subj="docTableHeaderField">
-    <span>Time <i ng-class="headerClass(indexPattern.timeFieldName)" ng-click="cycleSortOrder(indexPattern.timeFieldName)" tooltip="Sort by time"></i></span>
+    <span>Time <button
+      id="sort_by_time"
+      tabindex="0"
+      aria-label="{{sortText}}"
+      ng-class="headerClass(indexPattern.timeFieldName)"
+      role="button"
+      ng-click="cycleSortOrder(indexPattern.timeFieldName)"
+      tooltip="Sort by time"
+    ></button>
+    </span>
```

Comment:
Nice solution! A few lines below, I think we also need to update this markup similarly:

```html
    <span class="table-header-name">
      {{name | shortDots}} <i ng-class="headerClass(name)" ng-click="cycleSortOrder(name)" tooltip="{{tooltip(name)}}" tooltip-append-to-body="1"></i>
    </span>
```

I think this is what drives the appearance of the sort button for a non-time columns (e.g. `geo.srcdest`).  To show this column in the table header, just expand open a row and click this icon:

![image](https://user-images.githubusercontent.com/1238659/26897136-3840cde0-4b7c-11e7-843f-6606033a30cc.png)